src/main/kotlin/nmcp/internal/task/publish.kt

```diff
@@ -0,0 +1,198 @@
+package nmcp.internal.task
+
+import gratatouille.GInputFile
+import gratatouille.GLogger
+import gratatouille.GTask
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import nmcp.NmcpSpec
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okio.Buffer
+import okio.ByteString
+import org.gradle.api.Project
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.TimeSource.Monotonic.markNow
+
+
+@GTask(pure = false)
+fun publish(
+    logger: GLogger,
+    username: String?,
+    password: String?,
+    publicationName: String,
+    publicationType: String?,
+    baseUrl: String?,
```

Comment:
It might be fun to support `baseUrl: String = "https://central.sonatype.com/"` in Gratatouille. WDYT?

Reply:
I mean the following:

* Allow users to have code like `@GTask fun publish(baseUrl: String = "https://central.sonatype.com/")`
* Generate task wrapper class that uses `@Optional` for the given parameter
* Use the default value if the task property is not initialized. An alternative option could be `init { propertyName.convention(...defaultExpression)` yet I've heard you don't like conventions.

It would be nice to have both "non-nullable" and "declaration-site default value" when designing "task function".
At the same time, the current `val baseUrl = baseUrl ?: ...` looks awkward and prone to errors.

Reply:
```kotlin
@GTask fun publish(baseUrl: String = "https://central.sonatype.com/")
```

Yes! That would be pretty cool. But [KSP doesn't support reading the default parameter values](https://github.com/google/ksp/issues/642). So it's probably not doable in the current state of things.